### PR TITLE
grandpa: don't use block_on in Environment::report_equivocation

### PR DESCRIPTION
**Pull request by andresilva**
We can't actually use `block_on` because we already use it on the root future and it doesn't support recursive calls. Bug was introduced in #9128. Fixes https://github.com/paritytech/polkadot/issues/3322.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: client/finality-grandpa/src/environment.rs [hunk 1]
```diff
@@ -503,19 +503,19 @@ where
...
 		let is_descendent_of = is_descendent_of(&*self.client, None);
 
-		// TODO: add proper async support here
-		let best_header = futures::executor::block_on(
-			self.select_chain
-				.best_chain()
-				.map_err(|e| Error::Blockchain(e.to_string())),
-		)?;
+		let (best_block_hash, best_block_number) = {
+			// TODO: Use SelectChain::best_chain() to get a
```
Comment (kianenigma) [inline client/finality-grandpa/src/environment.rs:507]: will it need a linked issue?

Comment (kianenigma) [reply client/finality-grandpa/src/environment.rs:507]: also, is it important that these two pieces of data might be slightly outdated?

**Review by kianenigma — APPROVED** [review]
LGTM

File: client/finality-grandpa/src/environment.rs (hunk 1 above)
Comment (andresilva) [reply client/finality-grandpa/src/environment.rs]: I created an issue to track this and added a comment explaining the implications of the change here.

**Comment by andresilva** [comment]
I ran a local testnet with duplicate validators to generate an equivocation. Using this PR the panic is gone and the equivocation report was submitted successfully.

**Comment by andresilva** [comment]
bot merge

**Pull request merged**